### PR TITLE
Added warranty filter to Assets

### PR DESCRIPTION
**Pull request by GokulramGHV**
## Proposed Changes

Added `DateFromToRangeFilter` to the AssetFilter to filter assets by their warranty expiry date.

### Associated Issue

- Related to Coronasafe/care_fe#6241

## Merge Checklist

- [ ] Tests added/fixed
- [ ] Update docs in `/docs`
- [ ] Linting Complete
- [ ] Any other necessary step

_*Only PR's with test cases included and passing lint and test pipelines will be reviewed*_

@coronasafe/care-backend-maintainers @coronasafe/care-backend-admins

### CODE REVIEW

File: care/facility/tests/test_asset_api.py [hunk 1]
```diff
@@ -131,3 +130,29 @@ def test_asset_filter_in_use_by_consultation(self):
         response = self.client.get("/api/v1/asset/?in_use_by_consultation=false")
         self.assertEqual(response.status_code, status.HTTP_200_OK)
         self.assertEqual(len(response.data["results"]), 1)
+
+    def test_asset_filter_warranty_amc_end_of_validity(self):
+        asset1 = Asset.objects.create(
+            name="asset1",
+            current_location=self.asset_location,
+            warranty_amc_end_of_validity=datetime.now().date(),
```
Comment (Ashesh3) [inline care/facility/tests/test_asset_api.py:138]: - Use the linter to correct the failing lint check.
- Do not directly use naive datetime to create objects that expect a timezone aware datetime object, wrap this around Django's `make_aware` method. Example: `make_aware(datetime.now().date())`
- Or instead, here you can directly use `django.utils.timezone.now` to get the present aware datetime.

**Review by Ashesh3 — CHANGES_REQUESTED** [review]
- Use the linter to correct the failing lint check.
-

**Review by Ashesh3 — APPROVED** [review]

File: care/facility/api/viewsets/asset.py [hunk 2]
```diff
@@ -122,6 +122,9 @@ class AssetFilter(filters.FilterSet):
         method="filter_in_use_by_consultation"
     )
     is_permanent = filters.BooleanFilter(method="filter_is_permanent")
+    warranty_amc_end_of_validity = filters.DateFromToRangeFilter(
+        field_name="warranty_amc_end_of_validity"
+    )
```
Comment (rithviknishad) [inline care/facility/api/viewsets/asset.py:127]: is the `field_name` required in this case?

```suggestion
    warranty_amc_end_of_validity = filters.DateFromToRangeFilter()
```

Comment (Ashesh3) [reply care/facility/api/viewsets/asset.py:127]: Shouldn't be required, since the `field_name` and attribute name both are same. 🤔

**Review by rithviknishad — APPROVED** [review]

**Review by sainak — APPROVED** [review]

**Review by vigneshhari — APPROVED** [review]

**Pull request merged**